app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaApp.kt

```diff
@@ -31,93 +31,154 @@ import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.semantics.semantics
 import androidx.compose.ui.semantics.testTagsAsResourceId
+import androidx.lifecycle.compose.ExperimentalLifecycleComposeApi
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import androidx.navigation.NavDestination
 import androidx.navigation.NavDestination.Companion.hierarchy
+import com.google.samples.apps.nowinandroid.R
+import com.google.samples.apps.nowinandroid.core.data.util.NetworkMonitor
 import com.google.samples.apps.nowinandroid.core.designsystem.component.NiaBackground
 import com.google.samples.apps.nowinandroid.core.designsystem.component.NiaGradientBackground
 import com.google.samples.apps.nowinandroid.core.designsystem.component.NiaNavigationBar
 import com.google.samples.apps.nowinandroid.core.designsystem.component.NiaNavigationBarItem
 import com.google.samples.apps.nowinandroid.core.designsystem.component.NiaNavigationRail
 import com.google.samples.apps.nowinandroid.core.designsystem.component.NiaNavigationRailItem
+import com.google.samples.apps.nowinandroid.core.designsystem.component.NiaTopAppBar
 import com.google.samples.apps.nowinandroid.core.designsystem.icon.Icon.DrawableResourceIcon
 import com.google.samples.apps.nowinandroid.core.designsystem.icon.Icon.ImageVectorIcon
-import com.google.samples.apps.nowinandroid.core.designsystem.theme.NiaTheme
+import com.google.samples.apps.nowinandroid.core.designsystem.icon.NiaIcons
+import com.google.samples.apps.nowinandroid.feature.settings.R as settingsR
+import com.google.samples.apps.nowinandroid.feature.settings.SettingsDialog
 import com.google.samples.apps.nowinandroid.navigation.NiaNavHost
 import com.google.samples.apps.nowinandroid.navigation.TopLevelDestination
 
 @OptIn(
     ExperimentalMaterial3Api::class,
     ExperimentalLayoutApi::class,
-    ExperimentalComposeUiApi::class
+    ExperimentalComposeUiApi::class,
+    ExperimentalLifecycleComposeApi::class
 )
 @Composable
 fun NiaApp(
     windowSizeClass: WindowSizeClass,
-    appState: NiaAppState = rememberNiaAppState(windowSizeClass)
+    networkMonitor: NetworkMonitor,
+    appState: NiaAppState = rememberNiaAppState(
+        networkMonitor = networkMonitor,
+        windowSizeClass = windowSizeClass
+    ),
 ) {
-    NiaTheme {
-        val background: @Composable (@Composable () -> Unit) -> Unit =
-            when (appState.currentDestination?.route) {
-                TopLevelDestination.FOR_YOU.name -> { content ->
-                    NiaGradientBackground(content = content)
-                }
-                else -> { content -> NiaBackground(content = content) }
+    val background: @Composable (@Composable () -> Unit) -> Unit =
+        when (appState.currentDestination?.route) {
+            TopLevelDestination.FOR_YOU.name -> {
```

Comment:
I don't think this changed with this PR, but this comparison will no longer ever be true. I think it needs to be comparing with `currentTopLevelDestination`:

```suggestion
        when (appState.currentTopLevelDestination) {
            TopLevelDestination.FOR_YOU -> {
```